tvb_epilepsy/plot/plotter.py

```diff
@@ -0,0 +1,842 @@
+                          figure_name="", figure_dir=FOLDER_FIGURES, save_flag=SAVE_FLAG, show_flag=SHOW_FLAG,
+                          figure_format=FIG_FORMAT):
+        ax = self._prepare_distribution_axes(loc, scale, x, ax, linestyle, lgnd)
+        ax.set_title(distribution.type + " distribution")
+        self._save_figure(save_flag, pyplot.gcf(), figure_name, figure_dir, figure_format)
+        self._check_show(show_flag)
+        return ax, pyplot.gcf()
+
+    def _prepare_parameter_axes(self, parameter, x=numpy.array([]), ax=None, lgnd=True):
+        if ax is None:
+            _, ax = pyplot.subplots(1, 2)
+        if len(x) < 1:
+            x = linspace_broadcast(
+                numpy.maximum(parameter.low, parameter.scipy(parameter.loc, parameter.scale).ppf(0.01)),
+                numpy.minimum(parameter.high, parameter.scipy(parameter.loc, parameter.scale).ppf(0.99)), 100)
+        if x is not None:
+            plotter = Plotter()
+            ax[0] = plotter._prepare_distribution_axes(parameter, parameter.loc, parameter.scale, x, ax[0], "-", lgnd)
+            ax[0].set_title(parameter.name + ": " + parameter.type + " distribution")
+            ax[1] = plotter._prepare_distribution_axes(parameter, 0.0, 1.0, (x - parameter.loc) / parameter.scale,
+                                                       ax[1], "--", lgnd)
+            ax[1].set_title(parameter.name + "_star: " + parameter.type + " distribution")
+            return ax
+        else:
+            raise_value_error("Stochastic parameter's parameters do not broadcast!")
+
+    def plot_stochastic_parameter(self, parameter, x=numpy.array([]), ax=None, lgnd=True, figure_name="",
+                                  figure_dir=FOLDER_FIGURES,
+                                  save_flag=SAVE_FLAG, show_flag=SHOW_FLAG, figure_format=FIG_FORMAT):
+        ax = self._prepare_parameter_axes(parameter, x, ax, lgnd)
+        if len(figure_name) < 1:
+            figure_name = "parameter_" + parameter.name
+        self._save_figure(save_flag, pyplot.gcf(), figure_name, figure_dir, figure_format)
+        self._check_show(show_flag)
+        return ax, pyplot.gcf()
+
+    def plot_HMC(self, samples, skip_samples=0, title='HMC NUTS trace', figure_name=None, figure_dir=FOLDER_FIGURES,
+                 figsize=LARGE_SIZE, figure_format=FIG_FORMAT, show_flag=SHOW_FLAG, save_flag=SAVE_FLAG):
+        samples = ensure_list(samples)
+        if len(samples) > 1:
+            samples = list_of_dicts_to_dicts_of_ndarrays(samples)
+        else:
+            samples = samples[0]
+        nuts = extract_dict_stringkeys(samples, '__', modefun="find")
+        self.plots(nuts, shape=(2, 4), skip=skip_samples, xlabels={}, xscales={}, yscales={"stepsize__": "log"},
+                   title=title, figure_name=figure_name, figure_dir=figure_dir, figure_format=figure_format,
+                   figsize=figsize, show_flag=show_flag, save_flag=save_flag)
```

Comment:
This file is already large. 
Maybe split it into `PlotterStructures` and `PlotterResults` ?
Then use either through multiple inheritance by having `class Plotter(PlotterStructures, PlotterResults) ` or independently.